docs/devguide/contributing.rst

```diff
@@ -35,3 +35,71 @@ negative votes, the team will grant commit privileges.
 The core team will also be responsible for removing commit privileges when
 appropriate - for example for malicious merge behavior or just inactivity over
 an extended period of time.
+
+
+Quality gates for the pull requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are three quality gates to ensure the pull requests quality, `Hound`_ for
+code style check, `Travis CI`_ for unit-test, `Jenkins`_ for re-unit-test and
+functional test. When a pull request created, all tests will run automatically, test 
+results can be found in the merge status field of each pull request page.
+
+.. _Hound: https://houndci.com/
+.. _Travis CI: https://travis-ci.org/
+.. _Jenkins: https://jenkins.io/
+.. _jshint: http://jshint.com/
+
+**Hound**
+
+Hound works with `jshint`_ and comments on style violations in pull requests. 
+Configuration files ``.hound.yml`` and ``.jshintrc`` have been created in each 
+repository, so before creating a pull request, you can check code style locally with 
+jshint to find out style violations beforehand.
+
+**Travis CI**
+
+Travis CI integrates with unit test. The unit test scripts ``HWIMO-TEST`` have been 
+created in each repository, so before creating a pull request, you can run unit test 
+locally to find out basic errors beforehand.
+
+**Jenkins**
+
+Jenkins performs the re-unit-test and functional test then comments on the details of 
```

Comment:
It's probably more accurate to say:

Jenkins uses the Github Pull Request Builder plugin to monitoring all pull requests to perform quality gate tests prior to merge. The gates include running all the unit tests, running all dependent project unit tests with the code proposed from the pull request, and running an integration "smoke test" to verify basic end to end functionality.